### ISSUE TITLE
Auto scale down doesn't work when plan of address is changed

### ISSUE THREAD

**Issue by tkratky** [EnMasseProject/enmasse#1136]
**Description:**
When new address with broker.resource=2.0 is created and plan of this address is changed (in ConfigMap) to another plan with broker.resource=1.0 then StatefulSet is not scaled down to 1 replicas.

**Reproducer:** 
Reproducible via automated test in issue-1123-manual-scale-down branch (Will be soon disabled and merged with master)
`make SKIP_CLEANUP=true MULTITENANT=true systemtests SYSTEMTEST_ARGS="PlansTest#testMessagePersistenceAfterChangePlan"`

Output in standard-controller seems to be correct:
```
018-04-10 10:15:41 INFO  AddressController:107 - Total: 1, Active: 1, Configuring: 0, Pending: 0, Terminating: 0, Failed: 0
2018-04-10 10:15:41 INFO  AddressController:121 - Usage: {broker={distributed-queue={used=2.0, needed=2}}}, Needed: {broker={distributed-queue={used=2.0, needed=2}}}
2018-04-10 10:15:41 INFO  AddressController:153 - total: 160671818 ns, resolvedPlan: 60966 ns, calculatedUsage: 513646 ns, checkedQuota: 278396 ns, listClusters: 36928714 ns, provisionResources: 32433 ns, checkStatuses: 122754527 ns, deprovisionUnused: 74680 ns, replaceAddresses: 10601 ns, gcTerminating: 17855 ns
2018-04-10 10:16:04 INFO  FifoQueue:80 - Processing event Updated with key distributed-queue
2018-04-10 10:16:04 INFO  Reflector:55 - Waiting on event queue for 63818 ms unless notified
2018-04-10 10:16:04 INFO  AddressController:107 - Total: 1, Active: 1, Configuring: 0, Pending: 0, Terminating: 0, Failed: 0
2018-04-10 10:16:04 INFO  AddressController:121 - Usage: {broker={distributed-queue={used=1.0, needed=1}}}, Needed: {broker={distributed-queue={used=1.0, needed=1}}}
2018-04-10 10:16:04 INFO  AddressController:153 - total: 217535410 ns, resolvedPlan: 30964 ns, calculatedUsage: 347755 ns, checkedQuota: 244027 ns, listClusters: 101345708 ns, provisionResources: 32480 ns, checkStatuses: 115475770 ns, deprovisionUnused: 38836 ns, replaceAddresses: 10125 ns, gcTerminating: 9745 ns
2018-04-10 10:16:34 INFO  AddressController:107 - Total: 1, Active: 1, Configuring: 0, Pending: 0, Terminating: 0, Failed: 0
2018-04-10 10:16:34 INFO  AddressController:121 - Usage: {broker={distributed-queue={used=1.0, needed=1}}}, Needed: {broker={distributed-queue={used=1.0, needed=1}}}
2018-04-10 10:16:34 INFO AddressController:153 - total: 178032384 ns, resolvedPlan: 59036 ns, calculatedUsage: 1122613 ns, checkedQuota: 256846 ns, listClusters: 33028309 ns, provisionResources: 33349 ns, checkStatuses: 143277613 ns, deprovisionUnused: 222005 ns, replaceAddresses: 13230 ns, gcTerminating: 19383 ns
```

**Comment by lulf** [comment]
The issue here is that the standard-controller only acts on addresses in phase=Pending. This means that it will not attempt to check quota nor provision new resources for the address. I will try to change the REST API to change phase to Pending when an address is changed and see if that fixes it or causes havoc.

**Comment by lulf** [comment]
I think we might need another phase to handle this. If we set the phase to Pending, the agent will remove the configuration for the address from router and broker. 

Another alternative is to keep the Active phase, and find a way to 'detect' that plan has changed. For instance, if we add an annotation on the address called 'actual plan', we can detect that if it doesn't match the plan specified, we need to deal with plan change and then update the 'actual plan' annotation.

**Comment by lulf** [comment]
This is supported now and we have testd
